src/analysis/normed_space/box_subadditive.lean

```diff
@@ -0,0 +1,541 @@
+/-
+Copyright (c) 2020 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Yury G. Kudryashov
+-/
+import data.real.ennreal
+import topology.metric_space.basic
+import linear_algebra.affine_space.ordered
+import analysis.normed_space.add_torsor
+import analysis.specific_limits
+import analysis.asymptotics
+
+/-!
+# Sub/sup-additive functions on boxes
+
+Let `s` be a set in `ι → ℝ`. A subbox of `s` is a product of closed intervals which is included
+by `s`.
+
+A function `f : (ι → ℝ) → (ι → ℝ) → M` is called
+`box_subadditive_on`/`box_additive_on`/`box_supadditive_on` a set `s` if for any `l u : ι → ℝ`, `l ≤
+u`, `Icc l u ⊆ s` and any hyperplane `x i = c`, `l i ≤ c ≤ u i`, the sum of the values of `f` on the
+two subboxes `[l, u] ∩ {x | x i ≤ c}` and `[l, u] ∩ {x | c ≤ x i}` is greater than or
+equal/equal/less than or equal to its value on `[l, u]`.
```

Comment:
This docstring is hard to understand. Do you mean that you think of the value `f l u` as the value of the function on the box with lower corner `l` and upper corner `u`, i.e., the set `Icc l u`, that you denote with the usual interval notation `[l, u]`?